src/main/java/org/mariadb/jdbc/MariaDbDatabaseMetaData.java

```diff
@@ -86,12 +86,16 @@ public MariaDbDatabaseMetaData(Connection connection, String user, String url) {
     }
 
     /* Remove length from column type spec,convert to uppercase,  e.g  bigint(10) unsigned becomes BIGINT UNSIGNED */
-    static String columnTypeClause(String columnName) {
+    String columnTypeClause(String columnName) {
+        String parsedColumnName = parseColumnName(columnName);
+        return " UCASE(IF( " + parsedColumnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + parsedColumnName + ",1, LOCATE('(',"
+                + parsedColumnName + ") - 1 ), SUBSTRING(" + parsedColumnName + ",1+locate(')'," + parsedColumnName + "))), "
+                + parsedColumnName + "))";
+    }
 
```

Comment:
That's correct, but make request very big.
Would be better just to add just another IF, like : 

    static String columnTypeClause(int dataTypeMappingFlags, String columnName) {
        if ((dataTypeMappingFlags  & MariaSelectResultSet.TINYINT1_IS_BIT) > 0) {
            return " UCASE( "
                    + " IF(" + columnName + " = 'tinyint(1)', 'BIT', "
                    + " IF( " + columnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + columnName + ",1, LOCATE('(',"
                    + columnName + ") - 1 ), SUBSTRING(" + columnName + ",1+locate(')'," + columnName + "))), "
                    + columnName + ")))";
        }
        return " UCASE(IF( " + columnName + " LIKE '%(%)%', CONCAT(SUBSTRING( " + columnName + ",1, LOCATE('(',"
                + columnName + ") - 1 ), SUBSTRING(" + columnName + ",1+locate(')'," + columnName + "))), "
                + columnName + "))";
    }